bin/commands/install/index.ts

```diff
@@ -0,0 +1,123 @@
+/*
+  This program and the accompanying materials are made available
+  under the terms of the Eclipse Public License v2.0 which
+  accompanies this distribution, and is available at
+  https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as std from 'cm_std';
+import * as xplatform from 'xplatform';
+import * as common from '../../libs/common';
+import * as config from '../../libs/config';
+import * as fs from '../../libs/fs';
+import * as zosdataset from '../../libs/zos-dataset';
+import * as zosJes from '../../libs/zos-jes';
+
+export function execute(): void {
+
+    common.printLevel1Message("Install Zowe MVS data sets");
+
+    const runtimeEnv = std.getenv('ZWE_zowe_runtimeDirectory');
+
+    const zoweConfig = config.getZoweConfig();
+    let runtime = zoweConfig.zowe?.runtimeDirectory;
+    const prefix = zoweConfig.zowe.setup?.dataset?.prefix;
+    const jclHeaderCfg = zoweConfig.zowe.setup?.jcl?.header;
+    let jclHeaderJoined;
+
+    if (!runtime) {
+      runtime = runtimeEnv;
+    } else {
+      // We need clean path for xplatform.loadFileUTF8, otherwise will fail for e.g. /zowe/./files/SZWESAMP//ZWEINSTL
+      runtime = fs.convertToAbsolutePath(runtime);
+      if (runtime != runtimeEnv) {
+          common.printErrorAndExit(`Error ZWEL0105E: The Zowe YAML config file is associated to Zowe runtime "${runtime}", which is not same as where zwe command is located "${runtimeEnv}".`, undefined, 105);
+      } 
+    }
+
+    if (!prefix) {
+      common.printErrorAndExit(`Error ZWEL0157E: Zowe dataset prefix (zowe.setup.dataset.prefix) is not defined in Zowe YAML configuration file.`, undefined, 157); 
+    } 
+
+    if (Array.isArray(jclHeaderCfg)) {
+        jclHeaderJoined = jclHeaderCfg.join("\n");
+    } else {
+        jclHeaderJoined = jclHeaderCfg.toString();
+    }    
+    
+    const ZWEINSTL=`${runtime}/files/SZWESAMP/ZWEINSTL`;
+    const DATASETS = [ 'SZWEAUTH', 'SZWEEXEC', 'SZWELOAD', 'SZWESAMP' ];
+    const allowOverwrite = std.getenv("ZWE_CLI_PARAMETER_ALLOW_OVERWRITE") == 'true';
+    const dryRun = std.getenv("ZWE_CLI_PARAMETER_DRY_RUN") == 'true';
+    const existingDatasets: string[] = [];
+    let skipJCL = false;
+
+    for (let ds in DATASETS) {
+        if (zosdataset.isDatasetExists(`${prefix}.${DATASETS[ds]}`)) {
+            if (allowOverwrite === false) {
+                common.printMessage(`Warning ZWEL0301W: ${prefix}.${DATASETS[ds]} already exists and will not be overwritten. For upgrades, you must use --allow-overwrite.`);
+                skipJCL = true;
+            } else {
+                common.printMessage(`Warning ZWEL0300W: ${prefix}.${DATASETS[ds]} already exists. Members in this data set will be overwritten.`);
+                existingDatasets.push(`${prefix}.${DATASETS[ds]}`);
+            }
+        }
+    }
+
+    if (skipJCL) {
```

Comment:
Should we consider this?
`if (skipJCL && !dryRun) {`

With `--dry-run`, we can continue and print the JCL for user.

Reply:
👍